graphql/rules.lua

```diff
@@ -31,13 +31,16 @@ end
 function rules.fieldsDefinedOnType(node, context)
   if context.objects[#context.objects] == false then
     local parent = context.objects[#context.objects - 1]
+    if(parent.__type == 'List') then
+      parent = parent.ofType
+    end
     error('Field "' .. node.name.value .. '" is not defined on type "' .. parent.name .. '"')
```

Comment:
I think `Field "field" is not defined on type "[Parent]"` here would be better (brackets to indicate List type).